From: Ben Hutchings <ben@decadent.org.uk>
Date: Thu, 02 Jan 2020 17:27:20 +0000
Subject: hrtimer: Avoid ABI change in 3.16.80
Forwarded: not-needed

Commit 82a084a03d4b "hrtimer: Store cpu-number in struct
hrtimer_cpu_base" added a new member to hrtimer_cpu_base.  This
structure is visible to modules, though they shouldn't touch it.

The structure is only allocated in the core kernel (statically,
percpu) so it is safe to add a new member at the end.  Do that and
hide it from genksyms.

---
--- a/include/linux/hrtimer.h
+++ b/include/linux/hrtimer.h
@@ -180,7 +180,6 @@ enum  hrtimer_base_type {
  */
 struct hrtimer_cpu_base {
 	raw_spinlock_t			lock;
-	unsigned int			cpu;
 	unsigned int			active_bases;
 	unsigned int			clock_was_set;
 #ifdef CONFIG_HIGH_RES_TIMERS
@@ -193,6 +192,9 @@ struct hrtimer_cpu_base {
 	ktime_t				max_hang_time;
 #endif
 	struct hrtimer_clock_base	clock_base[HRTIMER_MAX_CLOCK_BASES];
+#ifndef __GENKSYMS__
+	unsigned int			cpu;
+#endif
 };
 
 static inline void hrtimer_set_expires(struct hrtimer *timer, ktime_t time)
